File: cast-extractor/src/main/java/com/libreshockwave/tools/format/ChannelNames.java
package com.libreshockwave.tools.format;

/**
 * Utility for getting channel names and labels.
 */
public final class ChannelNames {

    private ChannelNames() {}

    /**
     * Returns a human-readable name for a channel index.
     */
    public static String get(int channelIndex) {
        if (channelIndex < 6) {
            return switch (channelIndex) {
                case 0 -> "Tempo";
                case 1 -> "Palette";
                case 2 -> "Transition";
                case 3 -> "Sound 1";
                case 4 -> "Sound 2";
                case 5 -> "Script";
                default -> "Ch " + (channelIndex + 1);
            };
        }
        return "Ch " + (channelIndex - 5);
    }

    /**
     * Creates an array of channel labels for a score table.
     */
    public static String[] createLabels(int channelCount) {
        String[] labels = new String[channelCount];
        for (int i = 0; i < channelCount; i++) {
            labels[i] = get(i);
        }
        return labels;
    }
}

File: cpp/src/editor/format/ChannelNames.cpp
#include "libreshockwave/editor/format/ChannelNames.hpp"

namespace libreshockwave::editor::format {

std::string ChannelNames::get(int channelIndex) {
    if (channelIndex < 6) {
        switch (channelIndex) {
            case 0:
                return "Tempo";
            case 1:
                return "Palette";
            case 2:
                return "Transition";
            case 3:
                return "Sound 1";
            case 4:
                return "Sound 2";
            case 5:
                return "Script";
            default:
                return "Ch " + std::to_string(channelIndex + 1);
        }
    }
    return "Ch " + std::to_string(channelIndex - 5);
}

std::vector<std::string> ChannelNames::createLabels(int channelCount) {
    std::vector<std::string> labels;
    if (channelCount <= 0) {
        return labels;
    }
    labels.reserve(static_cast<std::size_t>(channelCount));
    for (int i = 0; i < channelCount; ++i) {
        labels.push_back(get(i));
    }
    return labels;
}

} // namespace libreshockwave::editor::format
